extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunnerConfig.java

```diff
@@ -403,6 +419,12 @@ public Builder withCapacity(@Min(0) @Max(Integer.MAX_VALUE) Integer capacity)
       return this;
     }
 
+    public Builder withTaskPayloadAsEnvVariable(Boolean taskPayloadAsEnvVariable)
```

Comment:
I don't see where this method is called. 
And also could we consolidate the two new config items to one? The current two configs are excluding each other thus can be misconfigured. Maybe like this:
| Config Name | Type | Description | Default | Required |
|-------------|------|-------------|---------|----------|
| `druid.indexer.task.taskPayloadHandlingMethod` | `String` | Determines how task payloads should be passed to K8s jobs. Use `ENV_VARIABLE` if you want task payloads to be passed as a compressed environment variable (suitable for payloads <= 1 MB). Use `PAYLOAD_MANAGER` if you plan on using task payloads > 1 MB. | `ENV_VARIABLE` | No |